File: frontend/src/services/appointmentService.js
import api from "./api";

const URL = "api/appointments";

const appointmentService = {
    getAllAppointments() {
        return api.get(URL);
    },

    getAppointmentById(id) {
        return api.get(`${URL}/${id}`);
    },

    createAppointment(appointment) {
        return api.post(URL, appointment);
    },

    updateAppointment(id, appointment) {
        return api.put(`${URL}/${id}`, appointment);
    },

    deleteAppointment(id) {
        return api.delete(`${URL}/${id}`);
    }
};

export default appointmentService;

File: backend/src/main/java/com/healthsphere/backend/service/AppointmentService.java
package com.healthsphere.backend.service;

import com.healthsphere.backend.dto.AppointmentRequest;
import com.healthsphere.backend.dto.AppointmentResponse;

import java.util.List;

public interface AppointmentService {

    AppointmentResponse createAppointment(AppointmentRequest request);

    List<AppointmentResponse> getAllAppointments();

    AppointmentResponse getAppointmentById(Long id);

    AppointmentResponse updateAppointment(Long id,
                                          AppointmentRequest request);

    void deleteAppointment(Long id);
}